spec-2025x.md.html

```diff
@@ -0,0 +1,1498 @@
+
+The appearance of the thin-film effect is controlled by the film's thickness $t_i$ and its IoR $\eta_i$.
+
+| Name     | Type  | Default | Range   | Description                                  |
+|----------|-------|---------|---------|----------------------------------------------|
+| $i$      | float | 0.0     | [0,1]   | Weight of the thin-film layer.               |
+| $d_i$    | float | 0.4     | [0,1]   | Thickness of the thin-film layer in microns. |
+| $\eta_i$ | float | 1.3     | [1,2.5] | Index of refraction of the thin-film layer.  |
+[Table [thin-film-parameters]: List of thin-film parameters.]
+
+![Figure [thin-film-parameters]: Metal spheres with thin-film coating.](img/params_thin_film_2025x.png)
+
+We model the thin-film effect by introducing a thin layer on top of our core material, as proposed by [#BB17]. This layer approximates iridescence by a physically plausible simulation, leveraging the principles of wave optics. Evaluating it effectively replaces our Fresnel term $F(cos\Theta, f_0, f_{90})$ (Section [Fresnel]) with a more intricate term $R(cos\Theta, \eta, d_i, \eta_i)$, the thin-film reflectance. The new term comprehensively considers all inter-reflections inside the thin-film layer, encompassing constructive and destructive interference effects.
+
+Based on the thin-film layer weight $i$, we blend the original Fresnel with the thin-film reflectance for the specular core components
+$$
+OpaqueCore(...) = \mathbf{M}_r (1-i)F(...) + \mathbf{M}_r i R(...)...
+$$
+
+Likewise for transmission
+$$
+TransparentCore(...) = \mathbf{M}_t (1-i)(1-sF(...)) + \mathbf{M}_r i (1-R(...))...
+$$
+
+!!!
+    Please note how the specular weight doesn't affect the thin-film reflectance. It's obvious in the transmission equation above, but also true for the reflection as it uses the actual base IoR instead of the $f_0$ term.
+
+The diffuse BSDF is defined as a blend of the energy correct diffuse BSDF $B_c$ as given in eqn. [b_c] and a new term $B_{c,i}$
+
+\begin{equation}
+\begin{aligned}
+B(\mathbf{v}, \mathbf{l};...) = (1-i)B_c(\mathbf{v}, \mathbf{l}; ...) + i\cdot B_{c,i}(\mathbf{v}, \mathbf{l}; ...)
+\end{aligned}
+\end{equation}
+
+with
+
+\begin{equation}
+\begin{aligned}
+B_{c,i}(\mathbf{v}, \mathbf{l};...) = B_u(\mathbf{v}, \mathbf{l}; ...)( 1 - max(1-E_i(\mathbf{v}\cdot\mathbf{n};...), 1-E_i(\mathbf{l}\cdot\mathbf{n};...))).
+\end{aligned}
+\end{equation}
+
+$E_i$ is the thin-film reflectance adjusted for energy preservation at higher roughness values:
+
+\begin{equation}
+\begin{aligned}
+&E_i(cos\Theta;  r_{uv}, \eta, d_i, \eta_i) = \\
+&\quad\quad\quad lerp\left(lum(R(cos\Theta; r_{uv}, \eta, d_i, \eta_i)), F_{m,avg}(f_{0,R12}, 1), 1 - (1-r_{uv}^2)^5\right)
+\end{aligned}
+\label{weight_scaling}
+\end{equation}
+
+where
+
+- $lum$ is the relative luminance of the colored thin-film reflectance.
```

Comment:
How is $lum$ computed? And nitpick: $lum$ renders strangely, looks identical to l times u times m. Maybe call it $l_i$.

Reply:
$l$ is already in use for translucency and light vector. How it is computed depends on the working color space of the renderer that implements it.

Reply:
Could provide an example for sRGB renderes, like in Eq. 73.